react/AppLinker/index.jsx

```diff
@@ -0,0 +1,131 @@
+/* global __TARGET__ */
+
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import {
+  checkApp,
+  startApp,
+  isMobileApp,
+  isMobile,
+  openDeeplinkOrRedirect
+} from 'cozy-device-helper'
+import { NATIVE_APP_INFOS } from 'cozy-ui/transpiled/react/AppLinker/native'
+
+import expiringMemoize from './expiringMemoize'
+
+const expirationDelay = 10 * 1000
+const memoizedCheckApp = expiringMemoize(
+  checkApp,
+  expirationDelay,
+  appInfo => appInfo.id
+)
+
+/**
+ * Render-props component that provides onClick/href handler to
+ * apply to an anchor that needs to open an app.
+ *
+ * Handles several cases:
+ *
+ * - On mobile app and other mobile app available
+ * - On web (not mobile)
+ * - On web mobile
+ *
+ * @example
+ * ```
+ * <AppLinker slug='banks' href='http://dalailama-banks.mycozy.cloud'>{
+ *   ({ onClick, href, name }) => {
+ *     <a href={href} onClick={onClick}>
+ *       Open { name }
+ *     </a>
+ *   }
+ * }</AppLinker>
+ * ```
+ */
+export class AppLinker extends React.Component {
+  state = {
+    isMobileAppAvailable: null
+  }
+
+  constructor() {
+    super()
```

Comment:
Nit: It works as is, but it's cleaner to call `constructor` and `super` with `props` as parameter. More info [here](https://overreacted.io/why-do-we-write-super-props/)